### PR TITLE
pkg/lua: model in kconfig

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

As the title says.

### Testing procedure

Green Murdock

### Issues/PRs references

None

### CODE REVIEW

File: pkg/lua/Kconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+config PACKAGE_LUA
+    bool "LUA language package"
+    depends on TEST_KCONFIG
+    depends on HAS_ARCH_32BIT
+
+    select PACKAGE_TLSF
+    select MODULE_PRINTF_FLOAT
+    select MODULE_LUA-CONTRIB
+
+    select MODULE_LIBC_GETTIMEOFDAY if !CPU_NATIVE
```
Comment (leandrolanzieri) [inline pkg/lua/Kconfig:17]: This is actually needed for native, according to `Makefile.dep`, also should be brought when `newlib_syscalls_default` is in
```suggestion
    select MODULE_LIBC_GETTIMEOFDAY if CPU_NATIVE || MODULE_NEWLIB_SYSCALLS_DEFAULT
```

File: pkg/lua/Kconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config PACKAGE_LUA
+    bool "LUA language package"
+    depends on TEST_KCONFIG
+    depends on HAS_ARCH_32BIT
```
Comment (leandrolanzieri) [inline pkg/lua/Kconfig:11]: There are some blacklisted architectures as well
```suggestion
    depends on HAS_ARCH_32BIT
    depends on !HAS_ARCH_MIPS32R2
    depends on !HAS_ARCH_RISCV
```

File: pkg/lua/Kconfig [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (leandrolanzieri) [inline pkg/lua/Kconfig:12]: In the makefile there seems to be an extra condition in `Makefile.dep` blacklisting the `picolibc` feature, but I think they meant something like:
```
depends on !MODULE_PICOLIBC
```
Not 100% sure though

File: pkg/lua/Kconfig (hunk 1 above)
Comment (leandrolanzieri) [reply pkg/lua/Kconfig:17]: For what I see, we should be able to get rid of the `MODULE_LIBC_GETTIMEOFDAY` definition in `Kconfig.newlib`, which is introducing unnecessary dependencies. `MODULE_NEWLIB_SYSCALLS_DEFAULT` defaults to `y`, so the extra select is redundant. @fjmolinas is this correct?

I'd propose:
```
diff --git a/sys/Kconfig.newlib b/sys/Kconfig.newlib
index 2f846e80e3..b1ba0e6a34 100644
--- a/sys/Kconfig.newlib
+++ b/sys/Kconfig.newlib
@@ -21,14 +21,6 @@ config MODULE_NEWLIB_SYSCALLS_DEFAULT
     help
         Default implementation of newlib system calls.
 
-config MODULE_LIBC_GETTIMEOFDAY
-    bool
-    select MODULE_NEWLIB_SYSCALLS_DEFAULT if MODULE_NEWLIB
-    select MODULE_XTIMER
-    select MODULE_ZTIMER64_XTIMER_COMPAT if MODULE_ZTIMER_XTIMER_COMPAT
-    help
-        Support for gettimeofday()
-
 endif # MODULE_NEWLIB
 
 config HAVE_CUSTOM_NEWLIB_SYSCALLS
```

**Review by MrKevinWeiss — APPROVED** [review]
Looks good and mudrock is happy!

**Comment by aabadie** [comment]
Thanks!

**Pull request merged**